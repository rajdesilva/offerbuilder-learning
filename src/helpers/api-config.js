import { init as initApm } from "@elastic/apm-rum";

export const baseURL = {
  API_URL_BASE: process.env.REACT_APP_BASE_API_URL,
  LOGIN_URL: process.env.REACT_APP_LOGIN_URL,
  LOGOUT_URL: process.env.REACT_APP_LOGOUT_URL,
  PROFILE_URL: process.env.REACT_APP_USER_PROFILE_URL,
  HISTORY_URL: process.env.REACT_APP_USER_HISTORY_URL,
};

export const apm = initApm({
  // Set required service name (allowed characters: a-z, A-Z, 0-9, -, _, and space)
  serviceName: process.env.REACT_APP_APM_SERVICE_NAME,
  serverUrl: process.env.REACT_APP_APM_SERVICE_URL,

  // Set service version (required for sourcemap feature)
  serviceVersion: process.env.REACT_APP_APM_SERVICE_VERSION,
});