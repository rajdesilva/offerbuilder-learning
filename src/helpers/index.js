export * from './api-config';
export * from './history';
