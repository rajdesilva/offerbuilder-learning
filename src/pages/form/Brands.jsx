import React, { Fragment, useState } from "react";
import { Row, Col, Form, Button, Spin, Divider } from "antd";
import { MinusCircleOutlined } from "@ant-design/icons";

import BrandConfig from "./BrandConfig";
import NemoSelect from "./NemoSelect";
import { useDispatch, useSelector } from "react-redux";
import { useRouteMatch } from "react-router-dom";
import { FormattedMessage } from "react-intl";
import { newOfferActions } from "../offers/actions";
import {
  checkIfMultipleBrandPresentForTarget,
  getListOfPropertiesForBrand,
  labelForBrands,
} from "../../helpers/utility";
import ConfirmRemoveSupplierModal from "./ConfirmRemoveSupplierModal";
import { supplySearchActions } from "../browseSupply/actions";
import styles from "./css/Brands.module.less";

function Brands(props) {
  const [removedObjectInfo, setRemovedObjectInfo] = useState({});
  const dispatch = useDispatch();
  const isBrowseSupplyPage = useRouteMatch({
    path: "/browse-supply/",
    strict: true,
    sensitive: true,
  });
  const { brands, loading, selectedProperties } = useSelector((state) => ({
    brands: window.getValue(state, `sourceinfo.brands`),
    loading: window.getValue(state, `sourceinfo.loading`),
    selectedProperties: window.getValue(state, "propertycart.cartItems"),
  }));
  const hideModal = () => {
    setRemovedObjectInfo({});
  };

  const removebrandBtn = () => {
    return (
      <Button
        type="text"
        data-testid="remove-source-btn"
        className={styles["brand-remove-btn"]}
        icon={<MinusCircleOutlined className={styles["minus-btn-icon"]} />}
        onClick={() => {
          if (props.isSettingsPage) {
            if (selectedProperties && selectedProperties.length > 0) {
              const isDuplicateBrandPresent =
                checkIfMultipleBrandPresentForTarget(
                  props.form.getFieldValue(["brands", props.field.key])
                );
              const propertiesForBrand = getListOfPropertiesForBrand(
                props.form.getFieldValue(["brands", props.field.key])
              );
              if (
                !isDuplicateBrandPresent &&
                propertiesForBrand &&
                propertiesForBrand.length > 0
              ) {
                setRemovedObjectInfo({
                  isBrand: true,
                  brandIndex: props.field.name,
                  properties: propertiesForBrand,
                  value: props.form.getFieldValue(["brands", props.field.key]),
                  removeBrand: () => {
                    props.remove(props.field.name);
                  },
                });
              } else {
                props.remove(props.field.name);
                dispatch({
                  type: newOfferActions.NEW_OFFER_REMOVE_BRAND,
                  payload: props.field.name, // index of the brand to be removed
                });
              }
            } else {
              props.remove(props.field.name);
              dispatch({
                type: newOfferActions.NEW_OFFER_REMOVE_BRAND,
                payload: props.field.name, // index of the brand to be removed
              });
            }
          } else {
            props.remove(props.field.name);
            dispatch({
              type: supplySearchActions.SEARCH_REMOVE_BRAND,
              payload: props.field.name, // index of the brand to be removed
            });
          }
        }}
      >
        {props.isForSearchResultsPage ? (
          <Fragment />
        ) : (
          <Fragment>
            &nbsp;
            <FormattedMessage id="nemo.removeThisSource" />
          </Fragment>
        )}
      </Button>
    );
  };

  return loading ? (
    <Row gutter={[8, 8]}>
      <Col span={4} className="col1-text"></Col>
      <Spin className={styles["brand-spinner"]} />
    </Row>
  ) : brands.length === 0 ? (
    <Row gutter={[8, 8]}>
      <Col span={4} className="col1-text"></Col>
      <FormattedMessage id="nemo.noBrandsFound" />
    </Row>
  ) : (
    <React.Fragment>
      {process.env.REACT_APP_HIDE_CLIENT_AND_STORE ? (
        <Row gutter={[8, 8]}>
          {!props.isForSearchResultsPage ? (
            <Col span={4} className="col1-text">
              {labelForBrands(props, isBrowseSupplyPage)}
            </Col>
          ) : (
            <Fragment />
          )}
        </Row>
      ) : (
        <Row gutter={[8, 8]}>
          {!props.isForSearchResultsPage ? (
            <Col span={4} className="col1-text">
              {labelForBrands(props, isBrowseSupplyPage)}
            </Col>
          ) : (
            <Fragment />
          )}
          <Col span={props.isForSearchResultsPage ? 18 : 4}>
            <Form.Item
              data-testid="brand"
              label={<FormattedMessage id="nemo.brand" />}
              name={[props.field.name, "name"]}
              colon={false}
              hidden={process.env.REACT_APP_HIDE_CLIENT_AND_STORE}
              className={styles["dynamic-form"]}
              rules={[
                {
                  required: !process.env.REACT_APP_HIDE_CLIENT_AND_STORE,
                  message: <FormattedMessage id="nemo.pleaseSelectBrand" />,
                },
              ]}
            >
              <NemoSelect
                data-testid="select-brand"
                mode=""
                id="search-supply-brand"
                optionsList={brands}
                defaultValue={{
                  name: { id: "idb", name: "Internal Demo Brand" },
                  storefronts: [
                    {
                      name: { id: "ids", name: "Internal Demo Storefront" },
                      suppliers: [
                        {
                          channels: [
                            {
                              id: "DEMO_OFFERBUILDER",
                              name: "DEMO_OFFERBUILDER",
                            },
                          ],
                          name: [{ id: "ntp", name: "NTP" }],
                        },
                      ],
                    },
                  ],
                }}
                onChange={(value) => {
                  if (props.isSettingsPage) {
                    dispatch({
                      type: newOfferActions.NEW_OFFER_UPDATE_BRAND,
                      payload: {
                        brandInfo: JSON.parse(value),
                        brandIndex: props.field.name,
                      },
                    });
                  } else {
                    dispatch({
                      type: supplySearchActions.SEARCH_UPDATE_BRAND,
                      payload: {
                        brandInfo: JSON.parse(value),
                        brandIndex: props.field.name,
                      },
                    });
                  }
                }}
              />
            </Form.Item>
          </Col>
          {props.isForSearchResultsPage && props.field.key >= 1 ? (
            <Col>{removebrandBtn()}</Col>
          ) : props.field.key >= 1 ? (
            removebrandBtn()
          ) : (
            <Fragment></Fragment>
          )}
        </Row>
      )}
      <BrandConfig
        data-testid="brand-config-2"
        {...props}
        isBrowseSupplyPage={isBrowseSupplyPage}
      />
      {process.env.REACT_APP_HIDE_CLIENT_AND_STORE ? (
        <Fragment />
      ) : (
        <Row gutter={[8, 8]}>
          <Col span={6} className="col1-text"></Col>
          <Col span={18}>
            <Divider />
          </Col>
        </Row>
      )}
      {removedObjectInfo && removedObjectInfo.value ? (
        <ConfirmRemoveSupplierModal
          removedObjectInfo={removedObjectInfo}
          hideModal={hideModal}
        />
      ) : (
        <Fragment />
      )}
    </React.Fragment>
  );
}

export default Brands;
