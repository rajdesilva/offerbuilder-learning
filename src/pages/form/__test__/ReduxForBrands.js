export const ReduxForBrands = {
  searchparams: {
    destination: {},
    los: 1,
    onlySupplier: false,
    target: {
      suppliers: [],
      channels: [],
    },
    brands: [
      {
        storefronts: [
          {
            id: "ids",
            name: "Internal Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
        id: "idb",
        name: "Internal Demo Brand",
      },
    ],
    dateRange: {
      startDate: "2020-09-18",
      endDate: "2020-11-24",
    },

    lcn: false,
    remainingCapitalPool: "",
  },
  channelinfo: {
    loading: false,
    channels: [
      {
        id: "AT_CORDIAL",
        name: "",
        type: "Portal",
        status: "HIDDEN",
      },
      {
        id: "india_channel",
        name: "indian channel",
        type: "Portal",
        status: "HIDDEN",
      },
    ],
  },
  sourceinfo: {
    loading: false,
    brands: [
      {
        id: "idb",
        name: "Internal Demo Brand",
        storefronts: [
          {
            id: "ids",
            name: "Internal Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids2",
            name: "Best Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids3",
            name: "PC Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
      },
      {
        id: "cdb",
        name: "Client Demo Brand",
        storefronts: [
          {
            id: "cds",
            name: "Client Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids2",
            name: "Best Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids3",
            name: "PC Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  },
};
