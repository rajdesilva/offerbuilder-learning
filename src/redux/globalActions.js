export const globalActions = {
  'SET_SUCCESS': 'SET_SUCCESS',
  'SET_ERROR': 'SET_ERROR',
  'SET_LANGUAGE': 'SET_LANGUAGE',
  'CLEAR': 'CLEAR',
}