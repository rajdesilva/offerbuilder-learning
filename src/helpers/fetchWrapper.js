import { message } from "antd";
import { createIntl, createIntlCache } from "@formatjs/intl";
import * as languages from "../langs/";
import { baseURL } from "./api-config";
import { history } from "./history";

// This is optional but highly recommended
// since it prevents memory leak
const cache = createIntlCache();

// Create the `intl` object
const intl = createIntl(
  {
    // Locale of the application
    // TODO:
    locale: "en",
    // Locale of the fallback defaultMessage
    defaultLocale: "en",
    messages: languages["en"],
  },
  cache
);
const defaultRequest = {
  credentials: "include",
  headers: {
    "Content-Type": "application/json",
  },
};
export const fetchWrapper = (url, payload) => {
  payload = {
    ...defaultRequest,
    ...payload,
  };
  return fetch(url, payload).then(
    (response) => {
      // Check that the response is valid and reject an error
      // response to prevent subsequent attempt to parse json
      if (!response.ok) {
        if (response.status === 500) {
          message.error(
            intl.formatMessage({
              id: "nemo.error500msg",
            })
          );
        } else if (response.status === 503) {
          message.error(
            intl.formatMessage({
              id: "nemo.error503msg",
            })
          );
        } else if (response.status === 504) {
          message.error(
            intl.formatMessage({
              id: "nemo.error504msg",
            })
          );
        } else {
          message.error(
            intl.formatMessage({
              id: "nemo.somethingWentWrongPleaseTryAgain",
            })
          );
        }
        return;
      }
      if (response.status === 403) {
        history.push("/access-denied");
      }
      return response.json ? response.json() : response;
    },
    (error) => {
      /**
       * user url, payload and error here for sending to the logger.
       */
      if (error instanceof TypeError) {
        window.location.href = baseURL.API_URL_BASE + "redirect";
      }
      console.error("fetch error : ", error);
    }
  );
};
