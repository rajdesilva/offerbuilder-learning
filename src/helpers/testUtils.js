import React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { render as rtlRender } from "@testing-library/react";
import { getValue } from "./utility";
import rootReducers from "../redux/rootReducers";
import * as languages from "../langs";
import { IntlProvider } from "react-intl";
const initialReducerState = {};
window.getValue = getValue;

window.google = {
  maps: {
    Marker: class {},
    Map: class {},
    LatLngBounds: class {},
    places: {
      Autocomplete: class {},
      AutocompleteService: class {},
      PlacesServiceStatus: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
      PlacesAutocomplete: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
    },

    MarkerClusterer: class {},
    Geocoder: class {},
  },
};
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

function render(
  ui,
  {
    initialState = initialReducerState,
    store = createStore(rootReducers, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <IntlProvider locale="en" messages={languages["en"]}>
          {children}
        </IntlProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { render };
