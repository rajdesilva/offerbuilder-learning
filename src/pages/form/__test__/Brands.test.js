import React from "react";
import Brands from "../Brands";
import { ReduxForBrands } from "./ReduxForBrands";
import { createStore } from "redux";
import { Form } from "antd";
import { MemoryRouter } from "react-router-dom";

import { act, render, fireEvent, cleanup } from "../../../helpers/testUtils";

afterEach(cleanup);

describe("Brands component test cases", () => {
  test("Brands components snapshot and nothing is selected in the dropdowns", () => {
    const props = {
      field: {
        key: 0,
        name: 0,
      },
    };
    const store = createStore(() => ({ ...ReduxForBrands }));
    const FormWrapper = (props) => {
      const [form] = Form.useForm();
      return (
        <MemoryRouter initialEntries={["/browse-supply/"]}>
          <Form form={form} initialValues={{}}>
            <Brands {...{ ...props, form }} />
          </Form>
        </MemoryRouter>
      );
    };

    const { container, queryAllByText } = render(<FormWrapper {...props} />, {
      store,
    });

    expect(container.firstChild).toMatchSnapshot();
    expect(queryAllByText("Please Select").length).toBe(3); // // REACT_APP_HIDE_CLIENT_AND_STORE brand is displayed
  });
  test("Brand component is rendered corrected using searchparams", () => {
    const props = {
      field: {
        key: 0,
        name: 0,
      },
    };
    const store = createStore(() => ({ ...ReduxForBrands }));
    const FormWrapper = (props) => {
      const [form] = Form.useForm();
      return (
        <MemoryRouter initialEntries={["/browse-supply/"]}>
          <Form form={form} initialValues={ReduxForBrands.searchparams}>
            <Brands {...{ ...props, form }} />
          </Form>
        </MemoryRouter>
      );
    };

    const { queryAllByText } = render(<FormWrapper {...props} />, {
      store,
    });

    expect(queryAllByText("Please Select").length).toBe(0);
  });
  test("Brand component - add and remove storefrontns using click of buttons", async () => {
    const props = {
      field: {
        key: 0,
        name: 0,
      },
    };
    const store = createStore(() => ({ ...ReduxForBrands }));
    const FormWrapper = (props) => {
      const [form] = Form.useForm();
      return (
        <MemoryRouter initialEntries={["/browse-supply/"]}>
          <Form form={form} initialValues={ReduxForBrands.searchparams}>
            <Brands {...{ ...props, form }} />
          </Form>
        </MemoryRouter>
      );
    };

    const { getAllByTestId, queryAllByText, getByTestId } = render(
      <FormWrapper {...props} />,
      {
        store,
      }
    );

    expect(getByTestId("add-storefront")).toBeInTheDocument();
    await act(async () => {
      await fireEvent.click(getByTestId("add-storefront"));
      await fireEvent.click(getByTestId("add-storefront"));
    });

    expect(queryAllByText("Please Select").length).toBe(6);
    await act(async () => {
      await fireEvent.click(getAllByTestId("remove-storefront")[0]);
      await fireEvent.click(getAllByTestId("remove-storefront")[0]);
    });
    expect(queryAllByText("Please Select").length).toBe(0);
  });
});
