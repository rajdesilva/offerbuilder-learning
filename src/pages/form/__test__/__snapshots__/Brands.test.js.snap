// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brands component test cases Brands components snapshot and nothing is selected in the dropdowns 1`] = `
<form
  class="ant-form ant-form-horizontal"
>
  <div
    class="ant-row"
    style="margin: -4px -4px 4px -4px;"
  >
    <div
      class="ant-col ant-col-4 col1-text"
      style="padding: 4px 4px 4px 4px;"
    />
    <div
      class="ant-col ant-col-4"
      style="padding: 4px 4px 4px 4px;"
    >
      <div
        class="ant-row ant-form-item"
        data-testid="brand"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required ant-form-item-no-colon"
            for="0_name"
            title=""
          >
            Client
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                class="ant-select ant-select-single ant-select-show-arrow"
                data-testid="nemo-select-search-supply-brand"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                      aria-autocomplete="list"
                      aria-controls="rc_select_TEST_OR_SSR_list"
                      aria-haspopup="listbox"
                      aria-owns="rc_select_TEST_OR_SSR_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="rc_select_TEST_OR_SSR"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-item"
                    title="Internal Demo Brand"
                  >
                    Internal Demo Brand
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="ant-row"
      style="margin: -4px -4px 4px -4px;"
    >
      <div
        class="ant-col ant-col-4 col-1-text"
        style="padding: 4px 4px 4px 4px;"
      />
      <div
        class="ant-col ant-col-4"
        style="padding: 4px 4px 4px 4px;"
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class="ant-form-item-required ant-form-item-no-colon"
              for="0_storefronts_0_name"
              title=""
            >
              Store
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-select ant-select-single ant-select-show-arrow"
                  data-testid="nemo-select-0search-supply-storefront"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-placeholder"
                    >
                      Please Select
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-col ant-col-7"
        style="padding: 4px 4px 4px 4px;"
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class="ant-form-item-required ant-form-item-no-colon"
              for="0_storefronts_0_suppliers_0_name"
              title=""
            >
              Supplier
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-select ant-select-multiple ant-select-show-search"
                  data-testid="nemo-select-0brw-search-supply"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                      style="width: 0px;"
                    >
                      <input
                        aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                      <span
                        aria-hidden="true"
                        class="ant-select-selection-search-mirror"
                      >
                        
                        Â 
                      </span>
                    </span>
                    <span
                      class="ant-select-selection-placeholder"
                    >
                      Please Select
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-col ant-col-7"
        style="padding: 4px 4px 4px 4px;"
      >
        <div
          class="ant-row ant-form-item undefined undefined"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class="ant-form-item-required ant-form-item-no-colon"
              for="0_storefronts_0_suppliers_0_channels"
              title=""
            >
              Channels Only For NTP
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-select ant-select-multiple ant-select-show-search"
                  data-testid="nemo-select-0brand-search-channel"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                      style="width: 0px;"
                    >
                      <input
                        aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                      <span
                        aria-hidden="true"
                        class="ant-select-selection-search-mirror"
                      >
                        
                        Â 
                      </span>
                    </span>
                    <span
                      class="ant-select-selection-placeholder"
                    >
                      Please Select
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="ant-btn ant-btn-link"
        type="button"
      >
        <span
          aria-label="plus-circle"
          class="anticon anticon-plus-circle"
          data-testid="add-storefront"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="plus-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M696 480H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
            />
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
          </svg>
        </span>
        <span>
           
        </span>
      </button>
    </div>
  </div>
  <div
    class="ant-row"
    style="margin: -4px -4px 4px -4px;"
  >
    <div
      class="ant-col ant-col-6 col1-text"
      style="padding: 4px 4px 4px 4px;"
    />
    <div
      class="ant-col ant-col-18"
      style="padding: 4px 4px 4px 4px;"
    >
      <div
        class="ant-divider ant-divider-horizontal"
        role="separator"
      />
    </div>
  </div>
</form>
`;
